### PR TITLE
Fix OpenCV on Apple silicon

### PR DESCRIPTION
**Pull request by andysan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

OpenCV doesn't currently build on Apple silicon due to checks for x86 features. Make those checks conditional on x86.

### CODE REVIEW

**Comment by fxcoudert** [comment]
Thanks @andysan !

This is fine, except with `Hardware::CPU.intel?` instead of `Hardware::CPU.arch == :x86_64` please (but still one commit per formula). I've stopped CI, since it will have to run again with that change.

**Comment by andysan** [comment]
> Thanks @andysan !
> 
> This is fine, except with `Hardware::CPU.intel?` instead of `Hardware::CPU.arch == :x86_64` please (but still one commit per formula). I've stopped CI, since it will have to run again with that change.

Done.

**Review by carlocab — APPROVED** [review]

File: Formula/opencv@2.rb [hunk 1]
```diff
@@ -67,7 +67,8 @@ def install
     # https://github.com/Homebrew/homebrew-science/issues/2302
     args << "-DCMAKE_PREFIX_PATH=#{py_prefix}"
 
-    args << "-DENABLE_SSE41=ON" << "-DENABLE_SSE42=ON" if MacOS.version.requires_sse42?
+    args << "-DENABLE_SSE41=ON" << "-DENABLE_SSE42=ON" \
+      if Hardware::CPU.intel? && MacOS.version.requires_sse42?
```
Comment (carlocab) [inline Formula/opencv@2.rb:71]: Did `brew audit` not let you put this in an `if` block?

**Review by fxcoudert — APPROVED** [review]

File: Formula/opencv@2.rb (hunk 1 above)
Comment (andysan) [reply Formula/opencv@2.rb:71]: Nope, it didn't like the fact that it was a one-line if block.

**Comment by jeroen** [comment]
❤️ 🚀 🏆

**Pull request closed without merging**